OpenRA.Mods.Common/Activities/Air/ReturnToBase.cs

```diff
@@ -135,10 +140,47 @@ public override Activity Tick(Actor self)
 				var nearestResupplier = ChooseResupplier(self, false);
 
 				if (nearestResupplier != null)
-					return ActivityUtils.SequenceActivities(
-						new Fly(self, Target.FromActor(nearestResupplier), WDist.Zero, aircraft.Info.WaitDistanceFromResupplyBase, targetLineColor: Color.Green),
-						new FlyCircle(self, aircraft.Info.NumberOfTicksToVerifyAvailableAirport),
-						this);
+				{
+					if (aircraft.Info.VTOL)
+					{
+						var distanceFromResupplier = (nearestResupplier.CenterPosition - self.CenterPosition).HorizontalLength;
+						var distanceLength = aircraft.Info.WaitDistanceFromResupplyBase.Length;
+
+						// If no pad is available, move near one and wait
+						if (distanceFromResupplier > distanceLength)
+						{
+							var randomPosition = WVec.FromPDF(self.World.SharedRandom, 2) * distanceLength / 1024;
+							var target = Target.FromPos(nearestResupplier.CenterPosition + randomPosition);
+
+							Queue(new Fly(self, target, WDist.Zero, aircraft.Info.WaitDistanceFromResupplyBase));
```

Comment:
Are you sure about using `Queue` here? I haven't looked at the code changes in detail, but I expect this (as opposed to `QueueChild`) would put these after any other player-queued activities, which is probably not what we want.

Reply:
Judging by #16246, you're right. I also think it would be better to wait for #16246 to be merged first so I only have to update once, though.